packages/flutter/test/widgets/table_test.dart

```diff
@@ -376,6 +376,33 @@ void main() {
     // If the above bug is present this test will never terminate.
   });
 
+  testWidgets('Calculating flex columns with small width deficit', (WidgetTester tester) async {
+    const SizedBox cell = SizedBox(width: 1, height: 1);
+    await tester.pumpWidget(
+      ConstrainedBox(
+        constraints: BoxConstraints.tight(const Size(600, 800)),
+        child: Directionality(
+          textDirection: TextDirection.ltr,
+          child: Table(
+            columnWidths: const <int, TableColumnWidth>{
+              0: FlexColumnWidth(1.0),
+              1: FlexColumnWidth(0.123),
+              2: FlexColumnWidth(0.123),
+              3: FlexColumnWidth(0.123),
+              4: FlexColumnWidth(0.123),
+              5: FlexColumnWidth(0.123),
+              6: FlexColumnWidth(0.123),
+            },
+            children: <TableRow>[
+              TableRow(children: List<Widget>.filled(7, cell)),
+              TableRow(children: List<Widget>.filled(7, cell)),
+            ],
+          ),
+        ),
+      ),
+    );
```

Comment:
Let's expect something here as well. Also, can you add a comment that indicate the behavior of this test? Something like it will be stuck in an infinite loop.

Reply:
I'm not sure what can be expected here. With my changes, `pumpWidget` is going to fail, since performLayout() will get interrupted. If `pumpWidget` was synchronous, I could add something like `expect(() => pumpWidget..., returnsNormally)`, but I don't know how to apply that to async tests.
I've added a comment that an assertion failure should be expected if the error occurs.

Reply:
you can do expect(tester.takeException(), null);

Reply:
Done.